parser.go

```diff
@@ -130,7 +130,7 @@ func (p *Parser) ParseUnverified(tokenString string, claims Claims) (token *Toke
 	// parse Header
 	var headerBytes []byte
 	if headerBytes, err = p.DecodeSegment(parts[0]); err != nil {
-		if strings.HasPrefix(strings.ToLower(tokenString), "bearer ") {
+		if len(tokenString) < 7 || strings.HasPrefix(strings.ToLower(tokenString[:7]), "bearer ") {
```

Comment:
Nitpicking, but now this returns an error message that the token should not contain "bearer", if the token is too short, which might be confusing. Maybe like this instead?
```suggestion
		if len(tokenString) >= 7 && strings.HasPrefix(strings.ToLower(tokenString[:7]), "bearer ") {
```

I wonder if we even need this check at all. If the token really contains "bearer", the base64 decode will fail anyway, so maybe we could get rid of this alltogether?

Reply:
> 

Remove them all and only keeps "could not base64 decode header" message is clearer and more efficiency, though it will lose some error information. I recommend this way.